dotCMS/src/main/java/com/dotmarketing/portlets/folders/business/FolderAPIImpl.java

```diff
@@ -517,6 +528,34 @@ private void _deleteChildrenAssetsFromFolder(Folder folder, User user, boolean r
 
 	}
 
+	/**
+	 * Verifies if a page is being used as a detail page for any content type
+	 * @param user
+	 * @param c
+	 * @param ident
+	 * @throws DotDataException
+	 * @throws LanguageException
+	 */
+	private void validateRelatedContentType(User user, Contentlet c, Identifier ident)
+			throws DotDataException, LanguageException {
+		ContentTypeAPI contentTypeAPI = APILocator.getContentTypeAPI(user);
+		List<ContentType> relatedContentTypes = contentTypeAPI.search("page_detail='" + ident.getId() + "'");
+		HTMLPageAssetAPI htmlPageAssetAPI = APILocator.getHTMLPageAssetAPI();
+		String uri = htmlPageAssetAPI.fromContentlet(c).getURI();
+		//Verifies if the page is related to any content type
+		if (relatedContentTypes != null && !relatedContentTypes.isEmpty()){
+            StringBuilder relatedPagesMessage = new StringBuilder();
+            relatedPagesMessage.append(UtilMethods.escapeSingleQuotes(LanguageUtil.get(user,
+                    "HTML-Page-related-content-type-delete-error")));
+
+            relatedPagesMessage.append(relatedContentTypes.stream()
+                    .map(t -> t.name() + " - Detail Page: " + uri)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 't' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)